ui/ecsconfig.py

```diff
@@ -480,25 +480,33 @@ def add_all():
         return None
 
     if l:
-        o("Available Storage Pool configurations:")
-        for sp_name in list_all():
-            o("\t{}".format(sp_name))
+        available_sp_configs = list_all()
+        if available_sp_configs is not None:
+            o("Available Storage Pool configurations:")
+            for sp_name in available_sp_configs:
+                o("\t{}".format(sp_name))
+        else:
+            o("No storage pool configurations are present.")
 
     if r:
         o('Storage Pools currently configured:')
         for sp_config in get_all():
             o("\t{}".format(sp_config['name']))
 
     if a:
-        n = None
-        conf.api_set_timeout(300)
-        conf.api_close()
-        conf.api_reset()
-        tasks = add_all()
-        #o(tasks)
-        conf.api_set_timeout(API_TIMEOUT)
-        conf.api_close()
-        conf.api_reset()
+        available_sp_configs = list_all()
+        if available_sp_configs is not None:
+            n = None
+            conf.api_set_timeout(300)
+            conf.api_close()
+            conf.api_reset()
+            tasks = add_all()
+            #o(tasks)
+            conf.api_set_timeout(API_TIMEOUT)
+            conf.api_close()
+            conf.api_reset()
+        else:
+            o('No storage pool configurations were provided in deploy.yml')
```

Comment:
The validator does that for us.  It's not an exception for those to not exist when we're just listing available configurations.